compat/src/main/scala-2.11/scala/collection/compat/TraversableOnceTupleExtensions.scala

```diff
@@ -0,0 +1,47 @@
+package scala.collection.compat
+
+import scala.collection.{immutable => i, mutable => m}
+
+class TraversableOnceKVExtensionMethods[K, V](private val self: TraversableOnce[(K, V)]) extends AnyVal {
+  def toImmutableSortedMap(implicit ordering: Ordering[K]): i.SortedMap[K, V] = {
```

Comment:
I’m on the fence. I think we should only try to provide the 2.13 syntax on 2.11 and 2.12, but we shouldn’t try to introduce new operations. Why can’t `to(immutable.SortedMap)` cross-compile?

Reply:
Because `to` returns `Coll[A]`, in the case of `SortedMap` it returns a `Iterable[(K, V)]`.

Reply:
Ah, I remember this one. Then I guess we have to write `immutable.SortedMap.from(List(1 -> "1").map(x => x))`

Reply:
Some ToCBF are missing, I will implement them and fix: https://github.com/scala/scala-collection-compat/issues/103

Reply:
There should be no CBF instance involved in that code. We should just enrich the `immutable.SortedMap` object with a `from` operation, like we previously did: https://github.com/scala/scala-collection-compat/blob/master/compat/src/main/scala-2.11_2.12/scala/collection/compat/package.scala#L36

Reply:
I see this is a good idea!

Reply:
This does not create an intermediate collection?

```
  implicit class IterableFactoryExtensionMethods[CC[X] <: GenTraversable[X]](private val fact: GenericCompanion[CC]) {
    def from[A](source: TraversableOnce[A]): CC[A] = fact.apply(source.toSeq: _*)
  }
```

Reply:
Yes, the `.toSeq` call does create an intermediate collection :-/